sklearn/ensemble/_hist_gradient_boosting/_binning.pyx

```diff
@@ -48,11 +57,18 @@ cdef void _map_num_col_to_bins(const X_DTYPE_C [:] data,
         int middle
 
     for i in prange(data.shape[0], schedule='static', nogil=True):
-        left, right = 0, binning_thresholds.shape[0]
-        while left < right:
-            middle = (right + left - 1) // 2
-            if data[i] <= binning_thresholds[middle]:
-                right = middle
-            else:
-                left = middle + 1
-        binned[i] = left
+
+        if isnan(data[i]):
+            # unkown values are mapped to first bin. For this to be correct,
+            # support_missing_values must have been correctly set at fit time.
```

Comment:
Maybe say which `fit`? I suppose you mean `_BinMapper`'s.

Reply:
It's both BinMapper and BaseHistGradientBoosting actually, since BaseHistGradientBoosting passes the info to BinMapper

Reply:
BTW, if you're having a hard time understanding the whole `support_missing_values` thing, that means my comments and my code aren't clear enough.
So definitely let me know what you think would help make it clearer!

Would it help if I wrote a short .md file explaining all this? I think this could be appropriate, sometimes comments just aren't enough.

Reply:
I'm not sure how we could handle these cases. It's probably not as challenging if instead of reading in the order in which changes are shown on github. Once I get through a bit, then it gets a bit clearer. For large PRs, I wouldn't mind having either a large-ish description of the PR, explaining in kinda detail what goes on, or better an .md file somewhere beside the code? That probably would make it easier to understand the code, but if it's in the repo, then it's also probably hard to maintain with future changes.